src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -196,11 +193,16 @@ public void removePerson(Person key) {
      * {@code key} must exist in the address book.
      */
     public void removeJob(Job key) {
-        Set<String> matches = key.getMatches();
+        String jobIdentifier = key.getIdentifier();
         for (Person person: persons) {
-            String personIdentifier = person.getIdentifier();
-            if (matches.contains(personIdentifier)) {
-                person.removeMatch();
+            if (person.getMatch().filter(match -> match.equals(jobIdentifier)).isPresent()) {
```

Comment:
I get what you are trying to do, it quite big brain tbh, but I think this might affect readability. Maybe it is better to break it up. 

Perhaps you could add guard clauses before this to make the happy flow more prominient.

check match.isPresent? continue (might not be needed due to the next statement?)
check !match.equals(Optional.of(jobID))? continue

// code to remove job from relevant contacts

Reply:
Changed!